File: web2-project/src/app/model/work-instruction.ts
import { Device } from "./device";

export enum InstructionStatus{
    Executed,
    Unexecuted
}

export class WorkInstruction {
    public instructionID: number;
    public device: Device;
    public description: string;
    public status: InstructionStatus;
    public workPlanID: number;

    constructor(device: Device, description: string, workPlanID: number){
        this.device = device;
        this.description = description;
        this.status = InstructionStatus.Unexecuted;
        this.workPlanID = workPlanID;
     }
}

File: Web2BackEnd/Web2BackEnd/Models/WorkInstruction.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Web2BackEnd.Models
{
	public class WorkInstruction
	{
		[Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
		public int InstructionID { get; set; }
		[Required]
		public Device Device { get; set; }
		[Required]
		public string Description { get; set; }
		[Required]
		public InstructionStatus Status { get; set; }
		[Required]
		public int WorkPlanID { get; set; }
	}

	public enum InstructionStatus
	{
		Executed,
		Unexecuted
	}
}
